lib/spreedly.ex

```diff
@@ -44,9 +44,9 @@ defmodule Spreedly do
 
   alias Spreedly.Environment
 
-  @spec add_gateway(Environment.t, String.t) :: {:ok, any} | {:error, any}
-  def add_gateway(env, gateway_type) do
-    post_request(env, add_gateway_path(), add_gateway_body(gateway_type))
+  @spec add_gateway(Environment.t, String.t, Map.t) :: {:ok, any} | {:error, any}
+  def add_gateway(env, gateway_type, gateway_params) do
```

Comment:
Hi @Thalagyrt instead of adding an additional function, can we update the existing `add_gateway` function head to accept `gateway_params` with a default of `%{}`?  Effectively this would move params/default params up a level.

Reply:
I didn't add an additional function, I added an additional parameter on the existing function such that we don't break API compatibility. I did miss the default empty map on that parameter though. To be clear before I do anything, are you advocating for a signature of `def add_gateway(env, gateway_params)` or a signature of `def add_gateway(env, gateway_type, gateway_params \\ %{})`? The former would break API compatibility while the latter would let us generate the gateway params as a map within `add_gateway/3` and change `add_gateway_body`'s signature to `def add_gateway_body(gateway_params)`.

Reply:
I misread the commit.

My suggestion is the latter - `def add_gateway(env, gateway_type, gateway_params \\ %{})` - this way we don't break backward compatibility. Then have `def add_gateway_body(gateway_type, gateway_params)` without the default for params.

Changing to `add_gateway_body/1` breaks from the approach we have in the other request body functions.

Reply:
I know it's not in the Readme yet, but can you fetch from upstream master and run `script/test` when you push your changes? If successful that will spit out a message that all formatting and tests passed - thanks in advance.

I'm planning to tag a new release once your latest changes are in place.

Reply:
Great - yeah, my intent was not to break compatibility but I derped up that method signature a bit. :)

I'll push a fix to my branch today and rebase against your latest master. I'm a bit tied up right now with getting our new payment service up and running, so it may be a few hours.

Cheers!